src/sentry/static/sentry/app/views/groupDetails/shared/utils.jsx

```diff
@@ -0,0 +1,34 @@
+import {Client} from 'app/api';
+
+/**
+ * Fetches group data and mark as seen
+ *
+ * @param {String} orgId organization slug
+ * @param {String} projectId project slug
+ * @param {String} groupId groupId
+ * @param {String} eventId eventId or "latest" or "oldest"
+ * @returns {Promise<Object>}
+ */
+export function fetchGroupEventAndMarkSeen(orgId, projectId, groupId, eventId) {
+  const api = new Client();
```

Comment:
Should the containing component pass the client in? That allows the component to cancel any pending API requests should it be unmounted.

Reply:
There's a `withApi` HoC